straxen/plugins/bayes_peaklet_classification.py

```diff
@@ -0,0 +1,138 @@
+import numpy as np
+import numba
+from scipy.special import logsumexp
+import strax
+import straxen
+
+
+class BayesPeakletClassification(strax.Plugin):
+    """
+    Bayes Peaklet classification
+    Returns the ln probability of a each event belonging to the S1 and S2 class.
+    Uses conditional probabilities and data parameterization learned from wfsim data.
+
+    :param peaks: peaks
+    :param waveforms: peaks waveforms in PE/ns
+    :param quantiles: quantiles in ns
+    :returns: the ln probability of a each peaklet belonging to S1 and S2 class
+    """
+    provides = 'peak_classification_bayes'
+    depends_on = ('peaks',)
+    __version__ = '0.0.1'
+    dtype = (strax.time_fields
+             + [('ln_prob_s1', np.float32, 'S1 ln probability')]
+             + [('ln_prob_s2', np.float32, 'S2 ln probability')]
+            )
+
+    # Descriptor configs
+    bayes_config_file = straxen.URLConfig(
+        default='resource://cmt://'
+                'bayes_model'
+                '?version=ONLINE&run_id=plugin.run_id&fmt=npy',
+        help='Bayes model, conditional probabilities tables and Bayes discrete bins'
+    )
+    bayes_n_nodes = straxen.URLConfig(
+        default=50,
+        help='Number of attributes(features) per waveform and quantile'
+    )
+    n_bayes_classes = straxen.URLConfig(
+        default=2,
+        help='Number of label classes S1(1)/S2(2)'
+    )
+
+    def setup(self):
+
+        self.class_prior = np.ones(self.n_bayes_classes)/self.n_bayes_classes
+        self.bins = self.bayes_config_file['bins']
+        self.cpt = self.bayes_config_file['cprob']
+
+    def compute(self, peaks):
+
+        waveforms, quantiles = compute_wf_and_quantiles(peaks, self.bayes_n_nodes)
+
+        ln_prob_s1, ln_prob_s2 = compute_inference(self.bins, self.bayes_n_nodes, self.cpt,
+                                                   self.n_bayes_classes, self.class_prior,
+                                                   waveforms, quantiles)
+
+        return dict(time=peaks['time'],
+                    endtime=peaks['time'] + peaks['dt'] * peaks['length'],
+                    ln_prob_s1=ln_prob_s1,
+                    ln_prob_s2=ln_prob_s2
+                    )
+
+def compute_wf_and_quantiles(peaks: np.ndarray, bayes_n_nodes: int):
+    """
+    Compute waveforms and quantiles for a given number of nodes(atributes)
+    :param peaks:
+    :param bayes_n_nodes: number of nodes or atributes
+    :return: waveforms and quantiles
```

Comment:
The waveforms are the down sampled versions of the original waveform right?

Reply:
yes